### PR TITLE
In engine_nightly_test.yml pass --version only if GITHUB_HEAD_REF is not empty

### PR DESCRIPTION
**Pull request by ptormene**
This should fix https://github.com/gem/oq-engine/actions/workflows/engine_nightly_test.yml

### CODE REVIEW

File: .github/workflows/engine_nightly_test.yml [hunk 1]
```diff
@@ -56,7 +61,12 @@ jobs:
         python-version:  ${{ matrix.python-version }}
     - name: Install dependencies
       run: |
-        python install.py devel --version=$GITHUB_HEAD_REF --repository=$GITHUB_REPOSITORY
+        if [[ "$GITHUB_HEAD_REF" != "" ]]
+        then
+          ipython install.py devel --version=$GITHUB_HEAD_REF --repository=$GITHUB_REPOSITORY
```
Comment (micheles) [inline .github/workflows/engine_nightly_test.yml:66]: Same question

File: .github/workflows/engine_nightly_test.yml [hunk 2]
```diff
@@ -31,7 +31,12 @@ jobs:
...
```
[the remark this answers is not in the document]
Comment (ptormene) [reply .github/workflows/engine_nightly_test.yml:36]: thanks, the i was inserted accidentally pressing 'i' to go into VIM insert mode, then the typo was copied also below

File: .github/workflows/engine_nightly_test.yml [hunk 3]
```diff
@@ -31,7 +31,12 @@ jobs:
...
+        else
+          $python install.py devel --repository=$GITHUB_REPOSITORY
```
[the remark this answers is not in the document]
Comment (ptormene) [reply .github/workflows/engine_nightly_test.yml]: For devel and devel_server, master is installed by default, with corresponding requirements

File: .github/workflows/engine_nightly_test.yml [hunk 4]
```diff
@@ -31,7 +31,12 @@ jobs:
         python-version:  ${{ matrix.python-version }}
     - name: Install dependencies
       run: |
-        python install.py devel --version=$GITHUB_HEAD_REF --repository=$GITHUB_REPOSITORY
+        if [[ "$GITHUB_HEAD_REF" != "" ]]
```
[the remark this answers is not in the document]
Comment (ptormene) [reply .github/workflows/engine_nightly_test.yml:34]: This is a very internal variable, used only by us to allow running tests also for PRs created in forks.

**Pull request merged**